src/test/test_utils.py

```diff
@@ -91,3 +91,154 @@ def test_get_target_coverage_url(coverage_url: str, fuzz_target: str, res: str,
     os.environ['FUZZ_INTROSPECTOR'] = "1"
     assert utils.get_target_coverage_url(coverage_url, fuzz_target, lang) == res
     del os.environ['FUZZ_INTROSPECTOR']
+
+
+@pytest.mark.parametrize(
+    ('cov_url', 'source_file', 'lineno', 'function_name', 'target_lang', 'temp_file', 'expect'),
+    [
+        (
+            'https://coverage-url.com/',
+            'fuzzlib/fuzzlib.c',
+            '13',
+            'name',
+            'c-cpp',
+            None,
+            'https://coverage-url.com/fuzzlib/fuzzlib.c.html#L13'
+        ),
+        (
+            'https://coverage-url.com/',
+            'Class',
+            '13',
+            'name',
+            'python',
+            None,
+            '#'
+        ),
+        (
+            'https://coverage-url.com/',
+            'Class',
+            '13',
+            'name',
+            'python',
+            '''{
+                 "format":2,
+                 "version":"6.5.0",
+                 "globals":"Test",
+                 "files":{
+                     "Test":{
+                         "hash":"Test",
+                         "index":{
+                             "relative_filename":"/src/fuzz_parse.py"
+                         }
+                     }
+                 }
+            }''',
+            '#'
+        ),
+        (
+            'https://coverage-url.com/',
+            'Class',
+            '13',
+            'fuzz_parse',
+            'python',
+            '''{
+                 "format":2,
+                 "version":"6.5.0",
+                 "globals":"Test",
+                 "files":{
+                     "Test":{
+                         "hash":"Test",
+                         "index":{
+                             "relative_filename":"/src/fuzz_parse.py"
+                         }
+                     }
+                 }
+            }''',
+            'https://coverage-url.com//Test.html#t13'
```

Comment:
Do we expect `...com//Test` here? I would assume we expect `...com/Test` here? Same in other tests

Reply:
Yes indeed, because both the java and python logic does add a seperator to it because the coverage link is supposed to be /covreport/linux ......

Reply:
this is a fail safe logic to ensure even if the coverage link does not ended with /, it still works.

Reply:
so which do we expect?

Reply:
URL sanitization to ensure there is only a single `/` is fixed in: https://github.com/ossf/fuzz-introspector/pull/661